### PR TITLE
Correct allOf with only one child schema (no discriminator)

### PR DESCRIPTION
**Pull request by wing328**
Correct allOf with only one child schema (no discriminator) by removing the old (incorrect) behavior. 

Now it works correctly as model composition.

cc @OpenAPITools/generator-core-team 

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/config/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by m-mohr** [comment]
How to do enable inheritance without discriminator?

**Comment by wing328** [comment]
Don't think you can do that according to the spec

> The OpenAPI Specification allows combining and extending model definitions using the allOf property of JSON Schema, in effect offering model composition. allOf takes an array of object definitions that are validated independently but together compose a single object.
>
> While composition offers model extensibility, it does not imply a hierarchy between the models. To support polymorphism, the OpenAPI Specification adds the discriminator field

Ref: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#composition-and-inheritance-polymorphism

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java [hunk 1]
```diff
@@ -1239,8 +1237,10 @@ public static String getParentName(ComposedSchema composedSchema, Map<String, Sc
...
                         // not a parent since discriminator.propertyName is not set
-                        hasAmbiguousParents = true;
-                        refedWithoutDiscriminator.add(parentName);
+                        // TOOD to be removed in 6.x release
+                        LOGGER.warn("[deprecated] inheritance without use of 'discriminator.propertyName' has been deprecated" +
+                                        " in the 5.x release. Composed schema name: {}. Title: {}",
+                                composedSchema.getName(), composedSchema.getTitle()); 
                     }
                 } else {
                     // not a ref, doing nothing, except counting the number of times the 'null' type
```
Comment (jeff9finger) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java:1246]: This block is no longer needed at all right? It only increments an unused variable `nullSchemaChildrenCount`

**Comment by wing328** [comment]
@jeff9finger I've pushed some update to minimize the change and only fix the incorrect behavior. Please have another look when you've time.

**Comment by typhoon2k** [comment]
Hi @wing328,

as far as I understand the only change applied here is change in log message content (and log level).
We have a very simple case:

base_schemas.yaml
```yaml
baseSchema:
  title: BaseSchema
  type: object
  description: Base schema description
  properties:
    property1:
      type: string
    property2:
      type: string
```

extension_schemas.yaml:
```yaml
extensionSchema
  title: ExtensionSchema
  type: object
  description: Extension schema description
  allOf:
    - $ref: 'base_schemas.yaml#/baseSchema'
  example:
    property1: 'value1'
    property2: 'value2'
```

With this definition (we have only one child schema here) we are still getting inheritance instead of composition and generation log contains a lot of WARN (previously)/INFO (with this update) messages.

So, behavior is not fixed for us...

best regards,
ty

**Comment by typhoon2k** [comment]
I've investigated this a bit firther and it seems that some generators are using own logic for allOf cases. In our case we are using typescript-axios and plantuml generators and both are generating it as inheritance (for example, typescript-axios generator is using special allOf template).
But what worries me is that we are getting many many "inheritance without use of 'discriminator.propertyName' has been deprecated in the 5.x release" messages in generator log. Do we need to produce this spam now?

best regards,
ty

**Comment by typhoon2k** [comment]
Hi @wing328 ,

what are the plans for this change? Can we get it in master sometime soon?

ty

P.S. It would be great to get rid of "inheritance without use of 'discriminator.propertyName' has been deprecated in the 5.x release" spamming message as well before merging this PR.

**Comment by jimschubert** [comment]
@typhoon2k I don't think we can add a breaking change and deprecation without informing the user.

@wing328 can you wrap the above message in `once` as other spammy logs in `ModelUtils`? This way, users could tweak logging by changing the system properties and defaults:

* `-Dorg.openapitools.codegen.utils.oncelogger.cachesize=200`
* `-Dorg.openapitools.codegen.utils.oncelogger.expiry=2000`

**Comment by typhoon2k** [comment]
@jimschubert , I just tried to play locally with once logger. I've made the following change to logging code:

```java
            once(LOGGER).info("[deprecated] inheritance without use of 'discriminator.propertyName' has been deprecated" +
                    " in the 5.x release. Composed schema name: " + composedSchema.getName() + ". Title: " + composedSchema.getTitle());
```

With this change I still get spam in logs. Do I need to configure once logger somehow additionally?

**Comment by jimschubert** [comment]
Excellent suggestion. I'll go ahead and merge this, then change the info log to be generic and add a debug log with the name and title.

**Comment by jimschubert** [comment]
@wing328 I'd merged this, but had to revert via #7323 as it caused all CI checks to fail in master. If you have a chance, could you merge master and take a look? It failed on `JavaClientCodegenTest#testAllowModelWithNoProperties` and I didn't readily see a way to allow for empty/marker classes on this change as the spec allows.

**Comment by typhoon2k** [comment]
hi @wing328 , @jimschubert , is there any chance this update will get into master sometime soon?

**Pull request merged**